13-1-1/recent_data.R

```diff
@@ -0,0 +1,154 @@
+# This file is called by compile_tables.R.
+# It processes the data about number of deaths caused by natural disasters in England
+# and Wales 2013-present along with the Age-Standardised mortality rate for those killed
+# by natural disasters.
+
+
+# Load in the datasets----
+
+# Includes data for all causes of death (non-disaster related)
+all_deaths <- read.csv(nomis_disaster_deaths_link) %>%
+  mutate(across(where(is.factor), as.character)) %>%
+  mutate(across(where(is.character), str_to_sentence)) %>% 
+  mutate(across(where(is.character), str_squish)) 
+
+all_mortality <- read.csv(nomis_mortality_link) %>% 
+  mutate(across(where(is.factor), as.character)) %>% 
+  mutate(across(where(is.character), str_to_sentence)) %>% 
+  mutate(across(where(is.character), str_squish)) 
+
+# Define some functions----
+
+get_GeoCode <- function(countryName){
+  if(countryName=="England") {return("E92000001")}
+  else if(countryName=="Wales") {return("W92000004")}
+  else if(countryName=="") {return("K04000001")}
+  else {return("")}
+}
+
+# Clean up all_deaths----
+
+all_deaths_cleaned <- all_deaths 
+all_deaths_cleaned <- all_deaths_cleaned %>% 
+  rename(Sex = GENDER_NAME, 
+         `Cause of death` = CAUSE_OF_DEATH_NAME, 
+         Value = OBS_VALUE, 
+         Country = GEOGRAPHY_NAME, 
+         Year = DATE) %>% 
+  mutate(`Observation status` = all_deaths$OBS_STATUS_NAME %>% unique()) %>%
```

Comment:
the pipe (%>%) means that it knows you are working from the all_deaths dataframe. You therefore don't need to tell it this on the right. This makes the code much cleaner as you can just put `mutate(`observation status` = OBS_STATUS_NAME) %>%`

Reply:
Because of the rewrite, the combining of dataframes before cleaning, the code still looks similar to this.

Reply:
That's fine Steven.